sdk/Dagger.SDK.Mod.SourceGenerator/SourceGenerator.cs

```diff
@@ -0,0 +1,124 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Dagger.SDK.Mod.SourceGenerator;
+
+/// <summary>
+/// Generate source code for running Dagger Function.
+/// </summary>
+[Generator(LanguageNames.CSharp)]
+public class SourceGenerator : IIncrementalGenerator
+{
+    private const string ObjectAttribute = "Dagger.SDK.Mod.ObjectAttribute";
+
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        context.RegisterPostInitializationOutput(postInitializationContext =>
+        {
+            postInitializationContext.AddSource("Dagger.SDK.Mod_Attributes.g.cs", ModuleAttributesSource());
+            postInitializationContext.AddSource("Dagger.SDK.Mod_Interfaces.g.cs", ModuleInterfacesSource());
```

Comment:
I'm current stuck with the unit tests because in unit test, it has no reference to the `Dagger.SDK.Mod`, cause it to fail because it cannot find any class related to our SDK. That's why I put it to generator.

Reply:
I think its difficult to convert to source generator once starting with reflection. It makes you need to change how to design/solution with TDD from the beginning